app/src/org/commcare/activities/LoginActivityUiController.java

```diff
@@ -221,9 +223,28 @@ public void refreshView() {
 
         // Decide whether or not to show the app selection spinner based upon # of usable apps
         ArrayList<ApplicationRecord> readyApps = MultipleAppsUtil.getUsableAppRecords();
+        if(ConnectManager.isConnectIdIntroduced()) {
+            //We need to remove any apps that are managed by Connect
+            String username = ConnectManager.getUser(activity).getUserId().toLowerCase(Locale.getDefault());
+            for(int i= readyApps.size()-1; i>=0; i--) {
+                String appId = readyApps.get(i).getUniqueId();
+                //Preset app needs to be in the list if set
+                if(!appId.equals(activity.getPresetAppId())) {
+                    AuthInfo.ProvidedAuth auth = ConnectManager.getCredentialsForApp(appId, username);
```

Comment:
abstract as ConnectManager.isConnectEnabledForApp()

Reply:
I left this out for now after abstracting the filterConnectManagedApps function. This would be a one-liner to check that the retrieved auth is not null, but I would still want to keep the other code out (i.e. retrieving Connect username) for simplicity.